perfkitbenchmarker/providers/azure/azure_network.py

```diff
@@ -21,264 +21,284 @@
-    _AzureEndpoint(vm.name, port, 'udp').Create()
-
-  def DisallowAllPorts(self):
-    """Closes all ports on the firewall."""
-    pass
+def GetResourceGroup():
+  """Get the resource group for the current benchmark."""
+  spec = context.GetThreadBenchmarkSpec()
+  if hasattr(spec, 'azure_resource_group'):
+    return spec.azure_resource_group
+  else:
+    group = AzureResourceGroup('pkb%s%s' % (FLAGS.run_uri, spec.uid))
+    spec.azure_resource_group = group
+    return group
 
 
-class AzureAffinityGroup(resource.BaseResource):
-  """Object representing an Azure Affinity Group."""
+class AzureResourceGroup(resource.BaseResource):
+  """A Resource Group, the basic unit of Azure provisioning."""
 
-  def __init__(self, name, zone):
-    super(AzureAffinityGroup, self).__init__()
+  def __init__(self, name):
+    super(AzureResourceGroup, self).__init__()
     self.name = name
-    self.zone = zone
+    # A resource group's location doesn't affect the location of
+    # actual resources, but we need to choose *some* region for every
+    # benchmark, even if the user doesn't specify one.
+    self.location = FLAGS.zones[0] if FLAGS.zones else DEFAULT_LOCATION
+    # Whenever an Azure CLI command needs a resource group, it's
+    # always specified the same way.
+    self.args = ['--resource-group', self.name]
 
   def _Create(self):
-    """Creates the affinity group."""
-    create_cmd = [AZURE_PATH,
-                  'account',
-                  'affinity-group',
-                  'create',
-                  '--location=%s' % self.zone,
-                  '--label=%s' % self.name,
-                  self.name]
-    vm_util.IssueCommand(create_cmd)
+    # A resource group can own resources in multiple zones, but the
+    # group itself needs to have a location. Therefore,
+    # FLAGS.zones[0].
+    _, _, retcode = vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'group', 'create',
+         self.name, self.location])
 
-  def _Delete(self):
-    """Deletes the affinity group."""
-    delete_cmd = [AZURE_PATH,
-                  'account',
-                  'affinity-group',
-                  'delete',
-                  '--quiet',
-                  self.name]
-    vm_util.IssueCommand(delete_cmd)
+    if retcode:
+      raise errors.RetryableCreationError(
+          'Error creating Azure resource group')
 
   def _Exists(self):
-    """Returns true if the affinity group exists."""
-    show_cmd = [AZURE_PATH,
-                'account',
-                'affinity-group',
-                'show',
-                '--json',
-                self.name]
-    stdout, _, _ = vm_util.IssueCommand(show_cmd, suppress_warning=True)
-    try:
-      json.loads(stdout)
-    except ValueError:
-      return False
-    return True
+    _, _, retcode = vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'resource', 'list', self.name])
+
+    return retcode == 0
+
+  def _Delete(self):
+    vm_util.IssueCommand(
```

Comment:
@voellm I think the desired end state is that PKB makes a graph of all resources it wants to allocate, then constructs a resource template for them and Azure allocates them all at once. I was trying to stick to PKB's normal deployment model until we could move to this option. Are you suggesting that I just not implement `_Delete` for Azure resources since the resource group will get them?